### PR TITLE
Enable Kubernetes RBAC

### LINKED ISSUE #680: Request Help to Specify --authorization-mode in API Model for k8s RBAC
**Is this a request for help?**:
yes
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
FEATURE REQUEST

k8s 1.6+ supports RBAC/ABAC. However, it requires the API server start with extra parameters, e.g. --authorization-mode=RBAC,ABAC --authorization-policy-file=mypolicy.json

Currently, there is no way to specify them in the API model.

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes 1.6+

**What happened**:
I'm unable to enable the RBAC for my cluster. 

**What you expected to happen**:
In API model, I can define the --authorization-mode or --authorization-policy-file for apiserver.

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:
You can specify multiple mode in the --authorization-mode, e.g. --authorization-mode=RBAC,ABAC. It's important for RBAC, since RBAC requires initial super user to setup the other roles.

### PR DESCRIPTION
**Pull request by lachie83**
This is the same work that's reference in #918

**What this PR does / why we need it**:
Activates RBAC - this seems to be the way forward with k8s.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Partially fixes #680 as it enables RBAC per default but does not help for upgrading existing clusters.

**Special notes for your reviewer**:
This haven't spent too much time to check that this is the most ideal way to set RBAC up, but it gets it working for now in our test cluster and thought I would share that work.

This depends on #909 which upgrades heapster to v1.4.0

I tested this on Azure on northeurope with a few of our services and seems to be all working.

## Testing
The following manual tests are run to determine if RBAC has been enabled and that all addon services are healthy.
* grep RBAC from /etc/kubernetes/manifests/kube-apiserver.yaml
* kubectl get clusterroles,clusterrolebindings (on 1.6+)
* kubectl get pods -n kube-system (confirm all are running
* kubectl logs <pod-name> -n kube-system and check for errors
* kubectl run busybox --image busybox -i -t nslookup kubernetes (test access to service discovery
* Create HPA https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/ (test heapster)
* kubectl proxy and click through UI https://localhost:8001/ui

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1143)

### CODE REVIEW

File: parts/kubernetesmaster-kube-addon-manager.yaml [hunk 1]
```diff
@@ -3,7 +3,6 @@ kind: Pod
 metadata:
   name: kube-addon-manager
   namespace: kube-system
-  version: v1
```
Comment (jackfrancis) [inline parts/kubernetesmaster-kube-addon-manager.yaml]: this is an unnecessary `version` metadata tag for `kube-addon-manager`, good change, let's put this in a separate PR

File: parts/kubernetesmasteraddons-heapster-deployment.yaml [hunk 2]
```diff
@@ -19,6 +76,10 @@ spec:
         scheduler.alpha.kubernetes.io/critical-pod: ""
         scheduler.alpha.kubernetes.io/tolerations: "[{\"key\":\"CriticalAddonsOnly\",\"operator\":\"Exists\"}]"
```
Comment (jackfrancis) [inline parts/kubernetesmasteraddons-heapster-deployment.yaml]: Given the additions below, let's get rid of this.

File: parts/kubernetesmasteraddons-heapster-deployment1.5.yaml [hunk 3]
```diff
@@ -0,0 +1,123 @@
+apiVersion: v1
```
Comment (jackfrancis) [inline parts/kubernetesmasteraddons-heapster-deployment1.5.yaml]: Note to self: this is a separate deployment manifest for 1.5 versions.

**Comment by jackfrancis** [comment]
Tested out O.K. on 1.5.3 and 1.6.6, LGTM!

**Comment by jackfrancis** [comment]
@acs-bot test this please

**Comment by travisnielsen** [comment]
Hi. For what it's worth, I tested this PR in my own ACS instance and ran into an issue related to the length of the customData field in azuredeploy.json. I'm deploying to a custom VNet.

`Deployment template validation failed: 'The template resource 'k8s-master-64230115-0' at line '1' and column '76718' is not valid: The language expression length limit exceeded. Limit: '24576' and actual: '25303'.. Please see https://aka.ms/arm-template-expressions for usage details.'.`

I got past it by deleting various comment lines.

**Comment by seanknox** [comment]
@acs-bot test this please

**Comment by seanknox** [comment]
@acs-bot test this please

**Comment by seanknox** [comment]
@lachie83 @jackfrancis The Calico e2e tests seem to be consistently failing still (on the last test run they were the only failures, actually). Do we need to create some additional RBAC roles?

**Comment by lachie83** [comment]
@sean. Yes hold on this and I'll get calico fixed although how is it being tested when RBAC is off by default. There have also been reports that the custom script is too large when enabling RBAC with other features.  Will need to investigate

**Comment by seanknox** [comment]
No worries—we'll keep kicking CI to re-run tests to get this over the hump. :)

**Comment by seanknox** [comment]
@acs-bot test this please

**Comment by seanknox** [comment]
@lachie83 please rebase off of master—we just merged in CircleCI support. Rebasing will kick off a job.
